vmsdk/python/tests/run.py

```diff
@@ -0,0 +1,38 @@
+"""
+Run tests for VMSDK.
+"""
+
+import argparse
+import logging
+import pytest
+from cctrusted_vm.cvm import ConfidentialVM
+
+LOG = logging.getLogger(__name__)
+
+def main():
+    """Run tests for VMSDK."""
+    parser = argparse.ArgumentParser(description="Run tests for VMSDK")
+    parser.add_argument(
+        "--no-cacheprovider",
+        default=False,
+        dest="no_cacheprovider",
+        help="No cacheprovider for pytest. True/False",
+        type=lambda x: (str(x).lower() in ['true','1', 'yes'])
+    )
+    args = parser.parse_args()
+    pytest_options = ["-v", "./"]
+    if args.no_cacheprovider is True:
+        pytest_options += ["-p", "no:cacheprovider"]
+
+    cc_type = ConfidentialVM.detect_cc_type()
```

Comment:
@dongx1x @ruomengh @intelzhongjie  do you have better approach to make filter in conftest? 
I know you already add mark.basic mark.tdx. But it looks wierd...

Reply:
I thought about this for a while. A rough idea on one alternative might be to specify the type of confidential VM we want to test and start that explicitly. It needs some change on the existing workflow (currently the test will be run in existing VM directly, we need to change this to that start VM and run the tests).

The current way leverages the existing running confidential VM as the action runner.  So it needs to detect the type of that existing VM where it runs to decide if it needs to run TDX specific tests.